iXBRLViewerPlugin/LocalViewer.py

```diff
@@ -0,0 +1,88 @@
+                    #if not file and relpath:
+                    #    print("relpath=" + relpath)
+                    if file == 'favicon.ico':
+                        return static_file("arelle.ico", root=cntlr.imagesDir, mimetype='image/vnd.microsoft.icon')
+                    _report, _sep, _file = file.partition("/")
+                    if _report.isnumeric(): # in reportsFolder folder
+                        if _file == "ixbrlviewer.js":
+                            root = os.path.join(os.path.dirname(__file__), "viewer", "dist")
+                            return static_file(_file, root=root, mimetype='text/javascript')
+                        else:
+                            root = reportsFolders[int(_report)]
+                        # print("file {} root {}".format(_file, root))
+                        # print (os.path.join(reportsFolders[int(_report)], _file))
+                        # is it an EDGAR workstation query parameter
+                        return static_file(_file, root=root,
+                                           # extra_headers modification to py-bottle
+                                           more_headers={'Cache-Control': 'no-cache, no-store, must-revalidate',
+                                                         'Pragma': 'no-cache',
+                                                         'Expires': '0'})
+                    return static_file(file, root="/") # probably can't get here unless path is wrong
+                except Exception as ex:
+                    cntlr.addToLog(_("iXBRLViewer local viewer exception: file: {} \nException: {} \nTraceback: {}").format(
+                        file, ex, traceback.format_tb(sys.exc_info()[2])), messageCode="localViewer:exception",level=logging.DEBUG)            
+            
+            # start server
+            localserver = Bottle()
+            
+            localserver.route('/<file:path>', 'GET', getlocalfile)
+            localserver.route('<relpath:path>', 'GET', getlocalfile)
+            # start local server on the port on a separate thread
+            threading.Thread(target=localserver.run, 
+                             # The wsgi server part of cherrypy was split into a new 'cheroot'
+                             #kwargs=dict(server='cherrypy', host='localhost', port=port, quiet=True), 
+                             kwargs=dict(server='cheroot', host='localhost', port=port, quiet=True), 
+                             daemon=True).start()
+            time.sleep(2) # allow other thread to run and start up
+    
+        localhost = "http://localhost:{}/{}".format(port, len(reportsFolders))
+        reportsFolders.append(reportsFolder)
+        cntlr.addToLog(_("iXBRLViewer local viewer: http://localhost:{}").format(port), messageCode="localViewer:listen",level=logging.DEBUG)
+        #cntlr.addToLog("localhost={}".format(localhost), messageCode="localViewer:listen",level=logging.DEBUG)
```

Comment:
Is this needed?

Reply:
Yes, but moved to arelle.LocalViewer in refactored module